connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -192,16 +192,20 @@
             + "/opt/connectors";
 
     public static final String CONFIG_PROVIDERS_CONFIG = "config.providers";
-    protected static final String CONFIG_PROVIDERS_DOC = "List of configuration providers. "
-            + "This is a comma-separated list of the fully-qualified names of the ConfigProvider implementations, "
-            + "in the order they will be created, configured, and used.";
+    protected static final String CONFIG_PROVIDERS_DOC =
+            "Comma-separated names of <code>ConfigProvider</code> classes, loaded and used "
+            + "in the order specified. Implementing the interface  "
+            + "<code>ConfigProvider</code> allows you to replace variable references in connector configurations, "
+            + "such as for externalized secrets. "
+            + "The implementations will be discovered using the standard Java <code>ServiceLoader</code> mechanism.";
```

Comment:
This last sentence is probably not for the user audience that will simply try to configure a worker. I suggest removing it.